aws-lc-rs/src/kdf.rs

```diff
@@ -0,0 +1,443 @@
+            SskdfDigestAlgorithmId::Sha256,
+            SskdfDigestAlgorithmId::Sha384,
+            SskdfDigestAlgorithmId::Sha512,
+        ] {
+            let alg = get_sskdf_digest_algorithm(id).expect("supported");
+            assert_eq!(id, alg.id());
+            assert_eq!(format!("{id:?}"), format!("{alg:?}"));
+            assert_eq!(format!("{id:?}"), format!("{:?}", alg.id()));
+            let mut output = vec![0u8; 32];
+            sskdf_digest(alg, &[1u8; 32], &[2u8; 32], &mut output).expect("success");
+        }
+    }
+
+    #[test]
+    fn sskdf_hmac_test() {
+        for id in [
+            SskdfHmacAlgorithmId::Sha224,
+            SskdfHmacAlgorithmId::Sha256,
+            SskdfHmacAlgorithmId::Sha384,
+            SskdfHmacAlgorithmId::Sha512,
+        ] {
+            let alg = get_sskdf_hmac_algorithm(id).expect("supported");
+            assert_eq!(id, alg.id());
+            assert_eq!(format!("{id:?}"), format!("{alg:?}"));
+            assert_eq!(format!("{id:?}"), format!("{:?}", alg.id()));
+            let mut output = vec![0u8; 32];
+            sskdf_hmac(alg, &[1u8; 32], &[2u8; 32], &[3u8; 32], &mut output).expect("success");
+        }
+    }
+
+    #[test]
+    fn kbkdf_ctr_hmac_test() {
+        for id in [
+            KbkdfCtrHmacAlgorithmId::Sha224,
+            KbkdfCtrHmacAlgorithmId::Sha256,
+            KbkdfCtrHmacAlgorithmId::Sha384,
+            KbkdfCtrHmacAlgorithmId::Sha512,
+        ] {
+            let alg = get_kbkdf_ctr_hmac_algorithm(id).expect("supported");
+            assert_eq!(id, alg.id());
+            assert_eq!(format!("{id:?}"), format!("{alg:?}"));
+            assert_eq!(format!("{id:?}"), format!("{:?}", alg.id()));
+            let mut output = vec![0u8; 32];
+            kbkdf_ctr_hmac(alg, &[1u8; 32], &[2u8; 32], &mut output).expect("success");
+        }
+    }
+
+    #[test]
+    fn algorithm_equality() {
+        let alg1 = get_kbkdf_ctr_hmac_algorithm(KbkdfCtrHmacAlgorithmId::Sha256).unwrap();
+        let alg2 = get_kbkdf_ctr_hmac_algorithm(KbkdfCtrHmacAlgorithmId::Sha256).unwrap();
+        assert_eq!(alg1, alg2);
+        let alg2 = get_kbkdf_ctr_hmac_algorithm(KbkdfCtrHmacAlgorithmId::Sha512).unwrap();
+        assert_ne!(alg1, alg2);
+
+        let alg1 = get_sskdf_digest_algorithm(SskdfDigestAlgorithmId::Sha256).unwrap();
+        let alg2 = get_sskdf_digest_algorithm(SskdfDigestAlgorithmId::Sha256).unwrap();
+        assert_eq!(alg1, alg2);
+        let alg2 = get_sskdf_digest_algorithm(SskdfDigestAlgorithmId::Sha512).unwrap();
+        assert_ne!(alg1, alg2);
+
+        let alg1 = get_sskdf_hmac_algorithm(SskdfHmacAlgorithmId::Sha256).unwrap();
+        let alg2 = get_sskdf_hmac_algorithm(SskdfHmacAlgorithmId::Sha256).unwrap();
+        assert_eq!(alg1, alg2);
+        let alg2 = get_sskdf_hmac_algorithm(SskdfHmacAlgorithmId::Sha512).unwrap();
+        assert_ne!(alg1, alg2);
+    }
+}
+
+#[cfg(test)]
+mod more_tests {
```

Comment:
These tests were previously to check behavior when "fips" was enabled, but they probably aren't needed now.